### DISCUSSION TITLE
Save state with .dib?

### DISCUSSION THREAD

**Discussion by dcuccia** [dotnet/interactive#1419] [Ideas]
Hello,

I really like the .dib format, as it is super clean and lets me easily version control the file without the embedded binary data.

That said, if I have run an expensive computation, I would like to be able to save and restore variable state in an adjacent binary format. This is a common pattern in Matlab. Is this currently built-in, or do I have to roll my own serialization/deserialization code?

Thanks!
David

**Comment by jonsequitur** [comment]
This isn't currently supported but it could certainly be a useful feature or candidate for an [extension](https://github.com/dotnet/interactive/blob/main/docs/extending-dotnet-interactive.md) package.

**Reply by dcuccia** [reply]
Thanks. I might be inclined to try and help, but would need some hand-holding. Interested to know if others might find this useful. What is the best path to move forward - shall I add as a feature request in Issues?
